sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -153,11 +156,19 @@ interface RequestBody {
 /**
  * Prepares the body before sending the request
  */
-function getRequestBody(body?: unknown, contentType: string = ""): RequestBody {
+function getRequestBody(
+  body?: unknown,
+  contentType: string = "",
+  options: InternalRequestParameters = {},
+): RequestBody {
   if (body === undefined) {
     return { body: undefined };
   }
 
+  if (options.skipSerialization) {
```

Comment:
Thank you for the suggestion. Discussed offline with @joheredi and it doesn't seem there is a particular reason for the ordering. The PR now updates to change the serialization order so that the content type check happens after `Uint8Array` check